pages/algorithms/url-normalisation.md

```diff
@@ -0,0 +1,15 @@
+---
+title: URL normalisation 
+key: url-normalisation 
+---
+
+A process to transform URLs in a standardised and consistent way, to make it possible to determine if two syntactically different URLs may be equivalent. (Source: [Wikipedia](https://en.wikipedia.org/wiki/URL_normalisation)).
```

Comment:
Is there something more stable than Wikipedia to refer to? For example, IETF and W3C specs will not change.

Reply:
Link is pointing to an article that does not exist on Wikipedia (at least not today). This is what @nitedog is referring to

Reply:
@nitedog, the normative references further down are good enough as stable references, I think. But the Wikipedia definition has such a nice way of explaining it. The link for Wikipedia was merely to give credit to the source. 
Any suggestions for how to handle this?

Reply:
The parsing URLs definition in the W3 HTML 5.2 spec defines this, by reference to the WHATWG URL standard:
https://www.w3.org/TR/html52/infrastructure.html#parsing-urls

This WHATWG parser deals with unicode normalisation, IDNs, resolving relative URLs, percent encoding and many backwards compatibility issues. The resulting serialized URLs can be reliably compared for identity. So this algorithm produces the same serialized URLs for all these cases

`http://www.example.com/about` base `http://www.example.com/`
`//www.example.com/about` base `http://www.example.com/`
`/about` base `http://www.example.com/`

**Note:** fully resolved URLs may be different, but still lead to the same page after resolving the HTTP request, due to redirects and DNS aliasing. For example:

```
http://example.com/
http://www.example.com/
https://www.example.com/
```

Reply:
Linking to the HTML URL parsing instead of defining anything myself.